db/crud.go

```diff
@@ -1044,6 +1044,104 @@ func (db *DatabaseCollectionWithUser) Put(ctx context.Context, docid string, bod
 	return newRevID, doc, err
 }
 
+func (db *DatabaseCollectionWithUser) PutExistingCurrentVersion(ctx context.Context, newDoc *Document, docHistory []string, docHLV HybridLogicalVector, existingDoc *sgbucket.BucketDocument) (doc *Document, cv *CurrentVersionVector, newRevID string, err error) {
```

Comment:
We don't expect callers of PutExistingCurrentVersion to be providing a revision history as well (docHistory). 

As mentioned on the ticket, this function should be computing a new revID for the mutation, similar to Put.